### PR TITLE
[IOPID-941] : Session Provider Refactoring

### PR DESCRIPTION
**Pull request by ClaudioNTT**
# Short Description
After the deployment, we noticed two issues.
- When attempting to change the locale from the URL route, it generated a redirect loop to **/ac**.
- The 404 page generated for static export of the website was the default one from Next.js and not the custom one.

👀 Please refer to this issue (regarding 404) at the following [link](https://github.com/vercel/next.js/issues/48227)

# List of Proposed Changes in this Pull Request
- The 404 page is being overwritten by the build script.
- Refactoring of the session provider component.
- Fixes on layout

# How to Test

1. yarn run build:prod
2. yarn run start-static

## First Scenario:

1. Go to localhost:3000
**2. You should expect to be redirected to /it/accedi (in local storage, you should have the 'it' locale)**

## Second Scenario:

1. Delete all data from local storage
2. Go to localhost:3000/pl (or whatever locale)
**3. You should expect to be redirected to the 404 page**

## Third Scenario:

1. Go to localhost:3000/it/whatever
**2. You should expect to be redirected to the 404 page**

## Fourth Scenario:
1. Adjust the callback URL for Hub SPID Login to match the trailing slash accordingly.
`ENDPOINT_ERROR=http://localhost:3000/it/accedi/errore/`
`ENDPOINT_SUCCESS=http://localhost:3000/it/accedi/`

3. Try to log in with Hub SPID login
**3. Check if, after login, you will see the profile page accordingly**

## Fifth Scenario:

1. When you are logged in
2. Go to localhost:3000/it/whatever
**3. You should expect to be redirected to the 404 page**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -5,7 +5,7 @@
   "scripts": {
     "dev": "next dev",
     "build": "copy .env .env.local && next build",
-    "build:prod": "next build",
+    "build:prod": "next build && cp -r out/it/404/index.html out/it/404/index.txt out/404/ ",
```
Comment (shadowsheep1) [inline package.json:8]:
```suggestion
    "build:prod": "next build && cp -f out/it/404/index.html out/it/404/index.txt out/404/ ",
```

`-r` usually stands for recursive, am I right? We don't need a recursive operation I guess.

**Review by shadowsheep1 — CHANGES_REQUESTED** [review]

File: src/app/[locale]/_utils/routes.ts [hunk 2]
```diff
@@ -35,3 +35,5 @@ export const PUBBLIC_ROUTES = [
 ];
 
 export const LOGIN_ROUTES = [ROUTES.LOGIN, ROUTES.LOGOUT_INIT];
+
+export const PRIVATE_ROUTES = EXISTING_ROUTES.filter((route) => !PUBBLIC_ROUTES.includes(route));
```
Comment (arcogabbo) [inline src/app/[locale]/_utils/routes.ts:39]: `PUBBLIC_ROUTES` typo here

**Review by arcogabbo — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (ClaudioNTT) [reply package.json]: Updated

**Review by shadowsheep1 — APPROVED** [review]
LGTM 🚀💯

| Scenario 1 | Scenario 3 | Scenario 3 | Scenario 4 | Scenario 5 |
| - | - | - | - | - |
| <video src="https://github.com/pagopa/io-web-profile/assets/16268789/701a1a9a-897b-48ee-8f00-1adef5f2f17c" > | <video src="https://github.com/pagopa/io-web-profile/assets/16268789/09d1a233-2d46-4942-92c1-46cff0fdebf9" > | <video src="https://github.com/pagopa/io-web-profile/assets/16268789/f359e289-7eba-4e0b-9b97-5fb248e9d611" > | <video src="https://github.com/pagopa/io-web-profile/assets/16268789/31a74402-3d7a-42e1-81ce-d5329f8b4ed1" > | <video src="https://github.com/pagopa/io-web-profile/assets/16268789/7d762d67-2eef-481c-b932-7f6cd3333fb0" >

**Review by arcogabbo — APPROVED** [review]

**Pull request merged**